modules/bridge/src/main/ts/ephox/bridge/components/view/ViewButton.ts

```diff
@@ -1,37 +1,107 @@
 import { FieldSchema, StructureSchema } from '@ephox/boulder';
-import { Result } from '@ephox/katamari';
+import { Optional, Result } from '@ephox/katamari';
 
 import * as ComponentSchema from '../../core/ComponentSchema';
 
-export interface ViewNormalButtonSpec {
-  type: 'button';
+interface BaseButtonSpec {
   text: string;
+  tooltip?: string;
   buttonType?: 'primary' | 'secondary';
-  onAction: () => void;
+  borderless?: boolean;
 }
 
-export type ViewButtonSpec = ViewNormalButtonSpec;
+export interface TogglableButtonApi {
+  isActive: () => boolean;
+  setActive: (state: boolean) => void;
+  setIcon: (newIcon: string) => void;
+}
 
-export interface ViewNormalButton {
+export interface ViewNormalButtonSpec extends BaseButtonSpec {
   type: 'button';
+  onAction: () => void;
+}
+
+export interface ViewTogglableIconButtonSpec extends Omit<BaseButtonSpec, 'text'> {
+  text?: string;
+  name: string;
+  type: 'togglableIconButton';
+  tooltip: string;
+  icon: string;
+  onAction: (api: TogglableButtonApi) => void;
+}
+
+export interface ViewButtonsGroupSpec {
+  type: 'group';
+  buttons: Array<ViewNormalButtonSpec | ViewTogglableIconButtonSpec>;
+}
+
+export type ViewButtonSpec = ViewNormalButtonSpec | ViewTogglableIconButtonSpec | ViewButtonsGroupSpec;
+
+interface BaseButton {
   text: string;
+  tooltip: Optional<string>;
   buttonType: 'primary' | 'secondary';
+  borderless: boolean;
+}
+
+export interface ViewNormalButton extends BaseButton {
+  type: 'button';
   onAction: () => void;
 }
 
-export type ViewButton = ViewNormalButton;
+export interface ViewTogglableIconButton extends Omit<BaseButton, 'text' | 'tooltip'> {
+  text: Optional<string>;
+  name: string;
+  type: 'togglableIconButton';
+  icon: string;
```

Comment:
Maybe better to put `icon` to BaseButton

Reply:
adding it in the base mean that we want to allow to use icon also in the normal button, is something that we want to do?

At the moment to have a button `icon` or `icon` + `text` at the moment we have `togglableIconButton`